src/App.tsx

```diff
@@ -13,6 +13,23 @@ export class App extends Component<{}, State> {
     movies: data,
   };
 
+  addMovie = (newMovie: Movie) => {
+    const { movies } = this.state;
+    let isAlreadyAdded = false;
+
+    movies.forEach(movie => {
+      if (movie.imdbId === newMovie.imdbId) {
+        isAlreadyAdded = true;
+      }
+    });
```

Comment:
you can use just method find. It's no need to check all array, enough to find just one same movie